src/foam/u2/crunch/wizardflow/CreateWizardletsAgent.js

```diff
@@ -38,55 +38,70 @@ foam.CLASS({
   ],
 
   methods: [
-    function execute() {
-      return Promise.all(this.capabilities
-        .reduce((updateUCJPromiseList, cap) => {
-          var associatedEntity, wizardlet;
-            
-          if ( Array.isArray(cap) && ( foam.nanos.crunch.MinMaxCapability.isInstance(cap[cap.length - 1]) ) ){
-            var minMaxCap = cap[cap.length - 1];
-
-            minMaxArray = [];
-
-            var choiceWizardlets = cap.slice(0, cap.length - 1).map(
-              capability => {
-                wizardlet = capability.wizardlet.cls_.create(
-                { 
-                  capability: capability, 
-                  isAvailable: false,
-                  ...capability.wizardlet.instance_ 
-                }, this)
+    function parseArrayToWizards(array, parentWizardlet){
+      var isOr = foam.nanos.crunch.MinMaxCapability.isInstance(array[array.length - 1]) ? true : false;
+      var updateUcjPromiseList = [];
 
-                associatedEntity = capability.associatedEntity === foam.nanos.crunch.AssociatedEntity.USER ? this.subject.user : this.subject.realUser;
+      var currentCap = array[array.length - 1];
+      var currentWizardlet  = isOr 
+        ? foam.nanos.crunch.ui.MinMaxCapabilityWizardlet.create({
+            capability: currentCap,
+            ...currentCap.wizardlet.instance_,
+            choiceWizardlets: []
+          }, this.__subContext__)
+        : currentCap.wizardlet.cls_.create(
+          { 
+            capability: currentCap, 
+            ...currentCap.wizardlet.instance_ 
+          }, this.__subContext__)
+      
+      array.slice(0, array.length - 1).forEach(
+        prereqCap => {
+          if ( Array.isArray(prereqCap) ){
+            updateUcjPromiseList = updateUcjPromiseList.concat(this.parseArrayToWizards(prereqCap, currentWizardlet));
+          } else {
+            var prereqWizardlet = prereqCap.wizardlet.cls_.create(
+              { 
+                capability: prereqCap, 
+                ...prereqCap.wizardlet.instance_ ,
```

Comment:
instance_ is meant to be an internal implementation detail, subject to change. Could you just not clone prereqCap.wizardlet the n set the capability? Wouldn't this accomplish the same thing more simply without exposing internals?

Reply:
Done.